dune_client/api/table.py

```diff
@@ -0,0 +1,98 @@
+"""
+Table API endpoints enables users to
+create and insert data into Dune.
+"""
+
+from __future__ import annotations
+from typing import List, Dict, Any, IO
+
+from dune_client.api.base import BaseRouter
+from dune_client.models import DuneError
+
+
+class TableAPI(BaseRouter):
+    """
+    Implementation of Table endpoints - Plus subscription only
+    https://docs.dune.com/api-reference/tables/
+    """
+
+    def upload_csv(
+        self,
+        table_name: str,
+        data: str,
+        description: str = "",
+        is_private: bool = False,
+    ) -> bool:
+        """
+        https://docs.dune.com/api-reference/tables/endpoint/upload
+        This endpoint allows you to upload any .csv file into Dune. The only limitations are:
+
+        - File has to be < 200 MB
+        - Column names in the table can't start with a special character or digits.
+        - Private uploads require a Plus subscription.
+
+        Below are the specifics of how to work with the API.
+        """
+        response_json = self._post(
+            route="/table/upload/csv",
+            params={
+                "table_name": table_name,
+                "description": description,
+                "data": data,
+                "is_private": is_private,
+            },
+        )
+        try:
+            return bool(response_json["success"])
+        except KeyError as err:
+            raise DuneError(response_json, "UploadCsvResponse", err) from err
+
+    def create_table(
+        self,
+        namespace: str,
+        table_name: str,
+        schema: List[Dict[str, str]],
```

Comment:
At some point we probably want to create an enum of the possible types, but not yet!

Reply:
Do you happen to have a list of all the possible types?

Reply:
Yup they're defined here 
https://dune-tables-docs.mintlify.app/api-reference/tables/endpoint/create

(Link is in docstring) 
![image](https://github.com/duneanalytics/dune-client/assets/28662355/7a22afe5-c557-45a9-8b38-c8efd2f157a1)